packages/testcontainers/src/generic-container/generic-container.test.ts

```diff
@@ -123,6 +123,19 @@ describe("GenericContainer", () => {
     expect(output).toEqual(expect.stringContaining("/tmp"));
   });
 
+  it("should set platform", async () => {
+    // This test uses an older version of cristianrgreco/testcontainer as all the other tests are using the latest version and the code doesn't pull the image again when the platform differs at this time
+    const container = await new GenericContainer("cristianrgreco/testcontainer:1.1.13")
+      .withWorkingDir("/tmp")
```

Comment:
Setting the work dir and command should have no effect on your test

Reply:
I have removed the `.withWorkingDir`-argument from the test case. I left the rest as that is an easy way to keep the container running so I can execute the `arch`-command to verify it pull the right arch version of the image.

I have locally confirmed it pulling the `linux/amd64`-version of the image on my Apple Silicon machine before making the pull policy change. See screenshot the below.

![CleanShot 2024-08-11 at 14 55 33@2x](https://github.com/user-attachments/assets/a5b15e0c-9e6e-49e9-b410-197096359b80)

Reply:
It'll work on your mac because you have rosetta emulation enabled, it won't work on ARM based linux/windows machines. Let's merge it anyway and if it's a big deal we'll think of something else

Reply:
Thank you. Right, didn't think of that. You would other ARM based environments Linux or Windows would have a similar solution or maybe QEmu works.